unity/unity-embed-host/CoreCLRHost.cs

```diff
@@ -18,6 +18,7 @@ static unsafe partial class CoreCLRHost
     static ALCWrapper alcWrapper;
     static FieldInfo assemblyHandleField;
     private static readonly ConcurrentDictionary<Assembly, AssemblyPair> m_assemblies = new ();
+    private static readonly object balanceLock = new();
```

Comment:
What does `balance` prefix indicate?

Reply:
Nit, I think you are safe to lock on `m_assemblies` as it's private and under your control.

Reply:
Gah got copiloted. Meant to rename.

Reply:
> Nit, I think you are safe to lock on `m_assemblies` as it's private and under your control.

I saw the established pattern is to have a new obj but yeah this is cleaner.